pkg/kube/util/url.go

```diff
@@ -61,13 +61,18 @@ func discoverURLFromRefs(client client.Client, ingressMonitor *endpointmonitorv1
 		return "", errors.New("No URL sources set for ingressMonitor: " + ingressMonitor.Name)
 	}
 
-	if urlFrom.IngressRef != nil && !kube.IsOpenshift {
+	if urlFrom.IngressRef != nil {
+		// if ingressRef is mentioned, it can be openshift or non openshift cluster
 		return discoverURLFromIngressRef(client, urlFrom.IngressRef, ingressMonitor.Namespace, ingressMonitor.Spec.ForceHTTPS, ingressMonitor.Spec.HealthEndpoint)
-	}
-	if urlFrom.RouteRef != nil && kube.IsOpenshift {
+
+	} else if kube.IsOpenshift && urlFrom.RouteRef != nil {
+		// if routeRef is mentioned in openshift cluster
 		return discoverURLFromRouteRef(client, urlFrom.RouteRef, ingressMonitor.Namespace, ingressMonitor.Spec.ForceHTTPS, ingressMonitor.Spec.HealthEndpoint)
+
 	}
 
-	log.V(1).Info("Unsupported Ref set on ingressMonitor: " + ingressMonitor.Name)
+	// if routeRef is mentioned in non openshift cluster
+	log.V(1).Info("RouteRef is not supported for kubernetes version: " + ingressMonitor.Name)
```

Comment:
can this be changed to warning?

Reply:
This should be Error. not Info. It should throw an ERROR as the monitor is expected to be created

Reply:
it is returning error below, we dont have to return error 2 times because it shows stacktrace which is messy